### ISSUE TITLE
Un-inject values from Unet

### ISSUE THREAD

**Issue by d8ahazard** [cloneofsimo/lora#9]
Right now, if we save_pretrained with the unet modifed for lora training, diffusion pipeline will refuse to load the unet later because of some "Linear" values.

Is there a way to un-inject the LORA values from unet before doing pipeline.save_pretrained?

**Comment by cloneofsimo** [comment]
Hmm... so as you mightve probably checked my code, there is [extract_lora_ups_down](https://github.com/cloneofsimo/lora/blob/ba349e56e23e92e3b128c7c67ae58d3067540daa/lora_diffusion/lora.py#L71) function that you can use to extract lora weights.
Now you can load new models, and merge lora with [weight_apply_lora](https://github.com/cloneofsimo/lora/blob/ba349e56e23e92e3b128c7c67ae58d3067540daa/lora_diffusion/lora.py#L106). Although that would take extra 10 seconds or so...

**Comment by cloneofsimo** [comment]
Not sure if this is the solution you were looking for, as it just extracts lora and reloads the new, raw model

**Comment by d8ahazard** [comment]
This is what I'm doing presently:

![image](https://user-images.githubusercontent.com/1633844/206865015-31c58b2b-9e5d-49ad-99b8-8355ea3a9a71.png)

In the above, I extract the lora weights, save it to a pt file, reload the unet from file, save the whole pipeline with the "clean" unet...

Then, I pass the path to my "compile checkpoint" method, which, when a .pt file is specified, will try to load the weights to the unet, save it separately, and then convert this new unet before compiling to a checkpoint.

But, as I said above...images generated with the pt + diffusers seem to work properly. Converting the same pipeline to a checkpoint results in images that have nothing to do with the trained subject...

![image](https://user-images.githubusercontent.com/1633844/206865107-6e188e83-48dc-4a0f-a51b-5f14fb4ab600.png)

**Comment by cloneofsimo** [comment]
So what you are doing is essentially what Im doing in `lora_add` cli, in mode `upl` (unet + lora). Can you check if that works for you?

**Comment by cloneofsimo** [comment]
```python
        loaded_pipeline = StableDiffusionPipeline.from_pretrained(
            path_1,
        ).to("cpu")

        weight_apply_lora(loaded_pipeline.unet, torch.load(path_2), alpha=alpha)

        if output_path.endswith(".pt"):
            output_path = output_path[:-3]

        loaded_pipeline.save_pretrained(output_path)
```

**Comment by cloneofsimo** [comment]
Seems pretty much the same logic for me, not sure why yours wouldn't work

**Comment by cloneofsimo** [comment]
By the way, nice work implementing to automatic's webui. I appreciate your work!

**Comment by d8ahazard** [comment]
Thanks!

Do you think it could be something in the conversion from diffusers to sd
that's causing issues?

I'll try the CLI, but as you said, I'm literally just using a slightly
modified version of that function to save the unet...

**Comment by cloneofsimo** [comment]
So, I think this issue is resolved via #18 ?

**Comment by tengshaofeng** [comment]
@d8ahazard @anotherjesse. I have add some code to support the Automatic111 WebUI. What you should do is just runing 'train_lora_dreambooth.py', then you will find that there is another lora_weight_webui.safetensors in the output dir. That is the right weight for WebUI.
reference my project in https://github.com/tengshaofeng/lora_tbq
